tests/test_cli_compile.py

```diff
@@ -1792,6 +1792,32 @@ def test_forwarded_args(PyPIRepository, runner):
     assert set(pip_args).issubset(set(args[0]))
 
 
+@pytest.mark.parametrize(
+    ("pip_args"),
+    (
+        ("--use-pep517", "--global-option=build_ext"),
+        ("--no-use-pep517", "--build-option=build_ext"),
+    ),
+)
+@mock.patch("piptools.scripts.compile.PyPIRepository")
+def test_forwarded_args_drop_unsupported(PyPIRepository, runner, pip_args):
+    """
+    Test the cli args (--pip-args 'arg...') are dropped when pip no longer supports them.
+    """
+    with open("requirements.in", "w"):
+        pass
+
+    cli_args = ("--no-annotate", "--generate-hashes")
+    runner.invoke(cli, [*cli_args, "--pip-args", " ".join(pip_args)])
+    args, kwargs = PyPIRepository.call_args
+    pip_current_version = get_pip_version_for_python_executable(sys.executable)
+    pip_breaking_version = Version("25.3")
+    if pip_current_version >= pip_breaking_version:
+        assert all([pip_arg not in args[0] for pip_arg in pip_args])
+    else:
+        assert all([pip_arg in args[0] for pip_arg in pip_args])
```

Comment:
Use generator expressions, no need for throwaway lists here:
```suggestion
    (first_posarg, *_tail_args), _kwargs = PyPIRepository.call_args
    pip_current_version = get_pip_version_for_python_executable(sys.executable)
    pip_breaking_version = Version("25.3")
    if pip_current_version >= pip_breaking_version:
        assert all(pip_arg not in first_posarg for pip_arg in pip_args)
    else:
        assert all(pip_arg in first_posarg for pip_arg in pip_args)
```

Reply:
Not sure I agree with this, performance wise it won't make a big difference as the pip_args list is never huge.
When a test fails, it outputs something like:
`assert False
 +  where False = all(<generator object test_forwarded_args_drop_unsupported.<locals>.<genexpr> at 0x0000024F32063300>)`

While, with the list it would be 
`assert False
 +  where False = all([False, True])`

Admittedly, it doesn't give a lot more info, but in any case it's a bit more clear as you can immediately see which index of the checked list is not correct.